simulator/src/main/java/simblock/node/consensus/ChiaProofOfSpace.java

```diff
@@ -0,0 +1,63 @@
+package simblock.node.consensus;
+
+import simblock.block.Block;
+import simblock.block.ChiaBlock;
+import simblock.block.PoSpaceFoliage;
+import simblock.block.PoSpaceTrunk;
+import simblock.node.Farmer;
+import simblock.node.Node;
+import simblock.simulator.Timer;
+import simblock.task.AbstractMintingTask;
+import simblock.task.VDFTask;
+
+import java.math.BigInteger;
+
+import static simblock.settings.SimulationConfiguration.VDF_MIN_TIME;
+import static simblock.settings.SimulationConfiguration.VDF_TIME_RANGE;
+
+public class ChiaProofOfSpace extends AbstractConsensusAlgo {
+    @Override
+    public AbstractMintingTask CreateMintingTask(Node node) {
+        if (node == null || node.getCurrentBlock() == null || !(node instanceof Farmer)) {
+            return null;
+        }
+
+        ChiaBlock unFinalizedBlock = this.createUnFinalizedBlock((Farmer) node);
+
+        // Should the delay be random like this or based on the quality?
+        // Should we add a check here to prevent blocks under a certain quality from being propagated?
+        // Do we need to propogate un-finalized blocks to the farmers? Is there any reason to?
+        long delay = VDF_MIN_TIME + (long) (Math.random() * VDF_TIME_RANGE);
+
+        return new VDFTask(node, delay, unFinalizedBlock, ((Farmer) node).getMintingBlock());
+    }
+
+    @Override
+    public boolean isReceivedBlockValid(Block receivedBlock, Block currentBlock) {
+        // NOTE: in PoW this method checks that the received block has height 0 or has a difficulty greater than
+        //       the parents nextDifficulty attribute - do we need to check for that in PoSpace?
```

Comment:
See comments - I think nextDifficulty is smth specific to bitcoin but not 100% sure